docs/testing/strategy.md

```diff
@@ -0,0 +1,253 @@
+
+## Integration Tests
+
+The integration tests are to test across a layer of the code base. In the case of Linero Reviewer we are only testing across the server as clinet will be handled with our acceptance tests. It is important that these are easy to run from both the developer machines and the CI and these should be run before any code is puched to the remote branch.
+
+### API Tests
+
+The purpose of API testing is to check that the server responds in a sensible way for a given input. API tests typically submit data to the server through a GraphQL mutation, and then check that the data was properly persisted by querying it again in a way similar to how to the client would request the data, and asserting on the response. This allows the team to recreate situations in the server-side that are difficult to reliably test with browser tests. It also checks that the API remains stable between client and server.
+
+#### Example tests
+
+To write a minimal example for a new resolver, you'd have to do something like the following:
+
+> Where the resololver takes a parameter of type `{a: string}` and returns an object with type `{b: number}`
+
+- first write the graphql definitions
+- Then in API-tests, you can do a `yarn graphql:codegen`, this will generate type definitions for all resolvers published by the application
+- Have a look in `tools/api-test/src/generates/graphql.ts`, and find the types for your new resolver
+- Write a resolver caller like this
+
+```ts
+import {
+  NewResolver,
+  NewResovlerOutput,
+  NewResolverInput,
+} from '../generated/graphql'
+import { ApiState } from './index'
+import { withAuthorisation } from '../utils'
+
+const newResolver = async (
+  ctx: ApiTestContext,
+  data: NewResolverInput,
+): Promise<{ newResolver: NewResovlerOutput }> => {
+  const query = `
+mutation NewResolver($data: NewResolverInput!) {
+ newResolver(data: $data) {
+   ...NewResolverOutput
+   __typename
+ }
+}
+`
+
+  // Call the graphql and return the result
+  return await withAuthorization(context, query, data)
+}
+
+const resolvers = { mutation: { newResolver } }
+
+export default resolvers
+```
+
+- Then in the test file (inside of `/tools/api-test/src/api-test/context.ts)
+
+```ts
+import { test, TestContext } from 'ava-ts'
+import context from '../xpub-api/context'
+import { defaultContext } from '../index'
+
+test('it does the thing', async (t: TestContext) => {
+  const context = defaultContext()
+  const result = await context.mutation.newResolver(context, {
+    a: 'some string',
+  })
+  t.deepEqual({ b: 100 }, result, 'it does the thing as expected')
+})
+```
+
+- This test should now run as part of the test suite
+
+#### Running it
+
+```bash
+# You need the platform running
+# pwd = elife-xpub/
+yarn start:services
+# You may need to setup the db
+yarn run pubsweet setupdb --clobber
+
+# Start the server inside of docker
+./scripts/run_app_in_docker.sh
```

Comment:
How about `yarn run app:docker` ... which is similar to `app:browser` ? .... This then stops the use of a script

Reply:
@petereast let's take this offline on another issue.